lib/plugin.js

```diff
@@ -0,0 +1,77 @@
+const fs = require('mz/fs');
+const JSZip = require('jszip');
+const path = require('path');
+const semver = require('semver');
+const Changelog = require('./changelog');
+
+function readInfosFromComposerJson(composerJsonString) {
+    const composerJson = JSON.parse(composerJsonString);
+
+    return {
+        version: composerJson.version,
+        shopwareCompatibility: composerJson.require['shopware/core'],
+        label: {
+            en: composerJson.extra.label['en-GB'],
+            de: composerJson.extra.label['de-DE'],
+        },
+    };
+}
+
+function readInfosFromPluginJson(pluginJsonString) {
+    const pluginJson = JSON.parse(pluginJsonString);
+    const compatibility = pluginJson.compatibility;
+
+    return {
+        version: pluginJson.currentVersion,
+        shopwareCompatibility: `>= ${compatibility.minimumVersion} && <= ${compatibility.maximumVersion}`,
+        label: pluginJson.label,
+    };
+}
+
+module.exports = class Plugin {
+    static async readFromZipFile(zipFilePath) {
+        if (!await fs.exists(zipFilePath)) {
+            throw new Error(`File ${zipFilePath} does not exist`);
+        }
+
+        const data = await fs.readFile(zipFilePath);
+        const zip = await JSZip.loadAsync(data);
+
+        const plugin = new Plugin();
+        const sw5RootFolder = zip.folder(
+            /^(Backend|Frontend|Core)\/\w+\//,
+        )[0];
+        const sw6RootFolder = zip.folder(
+            /^\w+\//,
+        )[0];
```

Comment:
I don't see a reason for breaking these statements across several lines. On another note: _folder_ is just a metaphor for directory.
```suggestion
        const sw5RootDirectory = zip.folder(/^(Backend|Frontend|Core)\/\w+\//)[0];
        const sw6RootDirectory = zip.folder(/^\w+\//)[0];
```